libs/community/langchain_community/document_loaders/pebblo.py

```diff
@@ -158,11 +160,16 @@ def _classify_doc(self, loading_end: bool = False) -> dict:
         for doc in doc_content:
             doc_metadata = doc.get("metadata", {})
             doc_authorized_identities = doc_metadata.get("authorized_identities", [])
-            doc_source_path = get_full_path(
-                doc_metadata.get(
-                    "full_path", doc_metadata.get("source", self.source_path)
+            if isinstance(self.loader, SharePointLoader):
+                doc_source_path = get_full_path(
```

Comment:
nitpick: If possible, let `get_full_path` handle the `full_path` computation instead of handling it in `_classify_doc`

Reply:
Computation i being done in `get_full_path` only, just added the condition here what to pass